docs/source/developers/guide/documentation.rst

```diff
@@ -27,4 +27,65 @@
 
 **************************
 Helping with documentation
-**************************
\ No newline at end of file
+**************************
+
+
+A great way to contribute to the project is to improve
+documentation. 
+
+If you are an Arrow user and you found some docs to be
+incomplete or inaccurate, share your hard-earned knowledge
+with the rest of the community.
+
+If you haven't come across an error in the documentation
+it is hard to find one and going through the docs in search
+for corrections is not a good way to go. In this case you can
+search for an issue that is dealing with the documentation
+on JIRA.
+
+.. note::
+   When searching for JIRA issue that deals with documentation
+   try selecting **Components** from **More** tab in JIRA search
+   and there find **Documentation** from the list.
+
+   See `Example search. <https://issues.apache.org/jira/issues/jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
```

Comment:
```suggestion
   See `Example search. <https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
```

(otherwise the link doesn't work for me)